PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/ui/WebAppSettingPanel.java

```diff
@@ -539,18 +539,28 @@ private void toggleRuntimePanel(final boolean isWindows) {
         cbLinuxRuntime.setVisible(!isWindows);
     }
 
+    private void toggleWebContainerSetting(boolean isDeployingWar) {
+        lblWebContainer.setVisible(isDeployingWar && rdoWindowsOS.isSelected());
+        cbWebContainer.setVisible(isDeployingWar && rdoWindowsOS.isSelected());
+    }
+
+    private void toggleDeployToRoot(final boolean isDeployingWar) {
+        chkToRoot.setVisible(isDeployingWar && isAbleToDeployToRoot());
+    }
 
-    // todo: separate the ToRoot and JarDeployHint toggle from this function
-    private void toggleWebContainerSetting(boolean isWar) {
-        lblWebContainer.setVisible(isWar && rdoWindowsOS.isSelected());
-        cbWebContainer.setVisible(isWar && rdoWindowsOS.isSelected());
-        chkToRoot.setVisible(isWar && isWebContainerAvailable());
-        lblJarDeployHint.setVisible(!isWar && rdoUseExist.isSelected());
+    private void toggleJarDeployHint(final boolean isDeployingWar) {
+        lblJarDeployHint.setVisible(!isDeployingWar && rdoUseExist.isSelected());
     }
 
-    private boolean isWebContainerAvailable() {
+    private boolean isAbleToDeployToRoot() {
```

Comment:
Why checking the OS to determine if is able to deploy to root instead of checking the deployment file.

Reply:
For Windows Web App, web container is always enabled. But for Linux Web App, Java SE web app has no web container defined.